### DISCUSSION TITLE
Tools for refit thermodata

### DISCUSSION THREAD

**Discussion by jiweiqi** [DENG-MIT/Arrhenius.jl#80] [Show and tell]
I guess this tool is able to refit the thermodata to a common T intersection, such that the if/else in the Thermo evaluation will be avoided.

Ref. https://github.com/Cantera/enhancements/issues/42#issuecomment-844657572

**Comment by jiweiqi** [comment]
@whitesides1 Apologize for bothering you here.

> We describe the methodology here [1], and the actual implementation is here [2].
> [1] https://doi.org/10.1016/j.combustflame.2020.06.010
> [2] https://github.com/LLNL/zero-rk/blob/master/applications/thermo_check/thermo_fix.cpp

I am wondering if the tool implemented in the website can help make the Tcommon in the Thermo data being consistent, say set it as 1000 K for all species.

**Reply by whitesides1** [reply]
@jiweiqi  The website is a front-end for the thermo_check application in the zero-rk repo.  The thermo_check application supports refitting to the global common temperature (with option fix_Tmatch [1]) however the web-site does not expose this option to users.

[1] https://github.com/LLNL/zero-rk/blob/b65a542d7bc4721ab4a882468022682e944ac909/applications/thermo_check/thermo_check.cpp#L255

**Reply by jiweiqi** [reply]
Hi @whitesides1 , got it, one can use the option of fix_Tmatch by using the C++ package directly. A following up question is there any plan to have a python-binding or Julia-binding for the feature. Based on my personal experience, install C++ is usually a painful experience with the painful process of installing all dependence.

**Reply by whitesides1** [reply]
We are considering a python-binding, but it would probably not remove the need to compile all the C++ code.  I understand well the pain of compiling "low-level" research codes, but we have tried to automate most of the process for Zero-RK.  Please try it out and post issues on the Zero-RK github page if you run into problems.

**Reply by jiweiqi** [reply]
Will definitely try it out. The functionality is very useful. It will be great if we can rely on Zero-RK for that functionality. We will see if users are good with using CMake.
